eyecite/find_citations.py

```diff
@@ -116,9 +106,9 @@ def get_citations(
 
 
 def extract_full_citation(
-    words: List[str],
+    words: Tokens,
     reporter_index: int,
```

Comment:
I feel like this should be renamed (also in the other `extract` functions), since this is no longer the index of the reporter itself (but also the volume, page, etc.). Maybe `token_index`? (Though that is a bit vague since `words` is a sequences of `Tokens`.) `matched_token_index`?

Reply:
Good point -- I just switched them all to `index`, as well as updating the docstrings. I didn't change `reporter_index` on the `Citation` object returned by those extract functions -- as I mentioned before I think we'll want to rethink what's exposed on the `Citation` object in general, so easier to fix it then.